File: api/playback.js
// In-memory playback state (per serverless instance — for production use
// Vercel KV, Upstash Redis, or a database)
let playbackState = {
  isPlaying: false,
  currentTrack: null,
  position: 0,
  duration: 0,
  queue: [],
};

export default function handler(req, res) {
  // CORS
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    res.status(204).end();
    return;
  }

  if (req.method === 'GET') {
    res.status(200).json(playbackState);
    return;
  }

  if (req.method === 'POST') {
    playbackState = { ...playbackState, ...req.body };
    res.status(200).json({ status: 'updated', state: playbackState });
    return;
  }

  res.status(405).json({ error: 'Method not allowed' });
}

File: api/playback.dart
import 'dart:io';
import 'dart:convert';

// In-memory playback state (would be replaced with a database in production)
Map<String, dynamic> _playbackState = {
  'isPlaying': false,
  'currentTrack': null,
  'position': 0,
  'duration': 0,
  'queue': [],
};

Future<void> handler(HttpRequest request) async {
  // CORS headers
  request.response.headers
    ..add('Access-Control-Allow-Origin', '*')
    ..add('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
    ..add('Access-Control-Allow-Headers', 'Content-Type');
  
  if (request.method == 'OPTIONS') {
    request.response.statusCode = HttpStatus.ok;
    await request.response.close();
    return;
  }
  
  if (request.method == 'GET') {
    request.response
      ..statusCode = HttpStatus.ok
      ..headers.contentType = ContentType.json
      ..write(jsonEncode(_playbackState));
    await request.response.close();
    return;
  }
  
  if (request.method == 'POST') {
    final body = await utf8.decoder.bind(request).join();
    try {
      final data = jsonDecode(body) as Map<String, dynamic>;
      _playbackState = {..._playbackState, ...data};
      request.response
        ..statusCode = HttpStatus.ok
        ..headers.contentType = ContentType.json
        ..write(jsonEncode({'status': 'updated', 'state': _playbackState}));
    } catch (e) {
      request.response
        ..statusCode = HttpStatus.badRequest
        ..headers.contentType = ContentType.json
        ..write(jsonEncode({'error': e.toString()}));
    }
    await request.response.close();
    return;
  }
  
  request.response.statusCode = HttpStatus.methodNotAllowed;
  await request.response.close();
}
